app/src/main/java/org/wikipedia/views/SurveyDialog.kt

```diff
@@ -0,0 +1,117 @@
+package org.wikipedia.views
+
+import android.app.Activity
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.textfield.TextInputEditText
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.analytics.eventplatform.PatrollerExperienceEvent
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.DimenUtil
+import org.wikipedia.util.FeedbackUtil
+
+object SurveyDialog {
+
+    fun showFeedbackOptionsDialog(activity: Activity, source: Constants.InvokeSource) {
+        sendAnalyticsEvent("toolbar_first_feedback", "pt_feedback", source)
+        var dialog: AlertDialog? = null
+        val feedbackView = activity.layoutInflater.inflate(R.layout.dialog_patrol_edit_feedback_options, null)
+
+        val clickListener = View.OnClickListener {
+            val feedbackOption = (it as TextView).text.toString()
+            dialog?.dismiss()
+            if (
+                (source == Constants.InvokeSource.PLACES &&
+                        feedbackOption != activity.getString(R.string.patroller_diff_feedback_dialog_option_unsatisfied) &&
+                        feedbackOption != activity.getString(R.string.places_survey_very_unsatisfied_option)) ||
+                (source != Constants.InvokeSource.PLACES && feedbackOption == activity.getString(R.string.patroller_diff_feedback_dialog_option_satisfied))
```

Comment:
Can we have the following format for a better reading experience? (one line per invokesource condition)
```
            if ((source == Constants.InvokeSource.PLACES && feedbackOption != activity.getString(R.string.patroller_diff_feedback_dialog_option_unsatisfied) && feedbackOption != activity.getString(R.string.places_survey_very_unsatisfied_option)) ||
                (source != Constants.InvokeSource.PLACES && feedbackOption == activity.getString(R.string.patroller_diff_feedback_dialog_option_satisfied))) {
```